### PR TITLE
Fix: py-adios Cython run

### PR DESCRIPTION
**Pull request by ax3l**
Always run Cython before `py-adios` installs. This makes sure the `.cpp` files from `.pyx` files are freshly created and work with newer CPython versions than the one checked in.

Related to #28003 (cc: @adamjstewart)

X-ref: https://github.com/ornladios/ADIOS/issues/216#issuecomment-999011323

Thanks @jychoi-hpc for the hint (I added you in the Spack package so you receive notifications about further updates).

### CODE REVIEW

**Review by jychoi-hpc — DISMISSED** [review]
Great. It looks good to me.

File: var/spack/repos/builtin/packages/py-adios/package.py [hunk 1]
```diff
@@ -60,10 +58,12 @@ def setup_file(self):
             return 'setup.py'
 
     def build_clib(self, spec, prefix):
-        # calls: make [MPI=y] python
-        args = ''
+        rm = which('rm')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-adios/package.py]: Let's use `os.remove` instead to make this more portable

**Review by adamjstewart — COMMENTED** [review]
Note that the `build_clib` phase will disappear and `setup_file` will no longer be used in #27798. You might want to check that PR as well to make sure things still work.

File: var/spack/repos/builtin/packages/py-adios/package.py (hunk 1 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/py-adios/package.py]: Sure! (although ADIOS1 is Unix only :D )

**Comment by ax3l** [comment]
https://github.com/spack/spack/pull/28223#pullrequestreview-843789717
> Note that the `build_clib` phase will disappear and `setup_file` will no longer be used in #27798. You might want to check that PR as well to make sure things still work.

As a follow-up to the current PR, I will probably need your help with that @jychoi-hpc . We essentially need to make sure that we can install the Python bindings via a local call to `python3 -m pip .`

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@ax3l I think we can make it work. For the `build_clib` stuff, we can use:
```python
@run_before('install')
def build(self):
    # Makefile stuff
```
For the `setup_file` stuff, we can rename `setup_mpi.py` to `setup.py` when `+mpi`.

**Comment by ax3l** [comment]
@adamjstewart thanks a lot! I see you added this already to #27798 - thanks a ton! :sparkles:

**Comment by ax3l** [comment]
@adamjstewart I tested and pushed one commit on your branch in #27798 now to make sure it works, just some small fixes to `py-adios` in it. Thanks a lot for your help :)

**Pull request merged**